.spellcheck.yaml

```
@@ -168,3 +167,13 @@ matrix:
         content: '[\s\S]*?'
         close: '^.. <!-- spellcheck-enable -->$'
   - pyspelling.filters.url:
+- name: Cpp
+  sources:
+  - ['*.cpp', '*.hpp']
```

By default, we should not check the cpp files. How can we make this happen?

---

In spellcheck, none of the sources can be empty. `expect_match = false` only works when it can't find files that matches the pattern specified. I suppose the only way to disable this would be to use some conditionals, but because we're using `rojopolis/spellcheck-github-actions`, this would increase the complexity of the work just to find a workaround. 

The simplest way was to put the above, as all other projects ensure they place C++ files in sub-directories and not the parent directory

---

I suggest then to only check cpp and hpp files in docs folder by default.
```suggestion
  - ['docs/**/*.cpp', 'docs/**/*.hpp']
```

---

That's a good compromise, added it in now